src/js/component.js

```diff
@@ -405,7 +406,9 @@ class Component {
     // Add the UI object's element to the container div (box)
     // Having an element is not required
     if (typeof component.el === 'function' && component.el()) {
-      this.contentEl().appendChild(component.el());
+      var childNodes = this.contentEl().children;
+      var refNode = childNodes[index] || undefined;
```

Comment:
~~wouldn't `refNode` end up pointing at the current element?~~ oh, I see now that it would be pointing at the current position in the elements list so it would be exactly the item we want to use here.

Also, it would automatically get the value `undefined` if `childNodes[index]` doesn't exist. It would be better to use `|| null`.